File: iniciante/python/1478.py
# -*- coding: utf-8 -*-

N = 1
while N != 0:
    N = int(input())
    if N == 0:
        break
    M = [[None for x in range(N)] for x in range(N)]
    for i in range(len(M)):
        aux = i+1
        for j in range(len(M)):
            if j < i:
                M[i][j] = aux
                aux -= 1
            elif j == i:
                aux = 1
                M[i][j] = aux
            else:
                aux += 1
                M[i][j] = aux
                
    for i in range(len(M)):
        for j in range(len(M)):
            if j == 0:
                print("%3d" %M[i][j],end="")
            else:
                print("%4d" %M[i][j],end="")
        print()
    print()

File: iniciante/c/1478.c
#include <stdio.h>
 
int main() {
   
    int N = 1, i, j,aux;
    do {
        scanf("%d", &N);
        if (N != 0){
            for (i=0; i <N; i++){
                for (j=0; j <N; j++){
                    if (j>0) {
                        printf(" ");
                    }
                    if (i == j){
                        printf("%3d", (i-j)+1);
                    }else{
                        if (j < i){
                            printf("%3d", (i+1)-(j));
                        }if ((j > i && (j -1)== i &&(j <= N-1))){
                            if (i == N-2 && j==N-1){
                                printf("%3d", 2);
                            }else{
                                printf("%3d", 2);
                                aux = 2;
                            }
                        }if ((j > i && (j-1)>i) &&  (j < N-1 && i < N-1)){
    if (i != N-2 && j != N-1){
                                printf("%3d", aux+1);
                                aux +=1;
        }
                        }if (j==N-1){
    if (i < N-2 && j <= N-1){
                            printf("%3d", aux+1);
        }
                        }
                    }
                }
                printf("\n");
            }
            printf("\n");
        }
       
    } while (N != 0);

    return 0;
}